### PR TITLE
New mobile menu and sync Hypha media-query breakpoints with tailwind

### LINKED ISSUE #3993: Remove old mobile menu and implement new solution

### PR DESCRIPTION
**Pull request by frjo**
Fixes #3993

## Test Steps

 - [ ] Confirm that the primary menu and header region works well on all screen sizes.

### CODE REVIEW

**Comment by wes-otf** [comment]
A few notes after testing:
- May be more personal preference but this feels a bit more cluttered to me than the previous mobile menu look, as staff the menu looks to take ~1/3rd the screen real estate. The applicant menu is less but still pretty significant. With the previous menu I liked that it was a bit more out of the way, as I don't think the user would necessarily need those options in every view.
- When not logged in, it looks like the nav bar is squeezed upwards a bit, might just need a little padding on the bottom.
Screenshot 2024-06-19 at 11 59 21

**Comment by frjo** [comment]
@wes-otf The narrow header when no menu is a bug I introduced, will fix that.

The mobile version of the  menu/header needs some work.

One option is to add a toggle to open/close the menu. When open we might show it fully expanded.

Other better ideas?

**Comment by wes-otf** [comment]
I like the concept of a menu toggle! seems like that could be a pretty intuitive option for the user

**Comment by frjo** [comment]
@theskumar Could you take a look at the mobile menu I created?

My idea is to have the menu as a plain list that you can toggle. I think that is going to work well. But my current implementation looks a bit … crude. Ideas for making it look and work better/nicer?

I'm rendering the menu twice, first the existing desktop version with alpine etc and then a mush more plain mobile version. Would prefer to only render it once but how to get a plain list version of it then?

**Comment by theskumar** [comment]
@frjo Yes. I can take a stab at it.

**Comment by theskumar** [comment]
@frjo @wes-otf  I've something working for it and but need some polish. I'm sharing some initial screenshots and let me know if they look good. Both mobile and desktop use the same html code. 


https://github.com/HyphaApp/hypha/assets/236356/408a0f50-8d38-48f6-95c4-1930ca7810ca

image
image
image

**Comment by frjo** [comment]
@theskumar This looks excellent!

**Comment by theskumar** [comment]
Pushed a working copy, will do some minor tweaking to it before finalizing it.

**Comment by wes-otf** [comment]
@theskumar this is really great, looks so clean!

File: hypha/core/templates/core/navigation/primarynav-apply.html [hunk 1]
```diff
@@ -1,71 +1,85 @@
...
-                                @click.outside="open = false"
-                                class="min-w-48 absolute block bg-white shadow-xl z-20 mt-4 border-y rounded-sm"
+<nav
+    aria-label="Primary"
+    x-show="showDesktopMenu || isDesktop"
+    class="fixed md:static top-0 md:top-auto end-0 bottom-0 bg-white z-20 md:z-10 shadow-xl md:shadow-none overflow-y-auto md:overflow-clip"
+>
+    <!-- Mobile Menu Opener -->
+    <button
+        class="absolute top-4 end-4 md:hidden border rounded p-1.5 z-30 hover:bg-slate-100 transition-colors"
```
Comment (frjo) [inline hypha/core/templates/core/navigation/primarynav-apply.html]: @theskumar Maybe make it `top-6` so the open and close buttons align a bit better?

File: hypha/core/templates/core/navigation/primarynav-apply.html [hunk 2]
```diff
@@ -1,71 +1,85 @@
...
+        @click="showDesktopMenu = false"
+    >
+        {% heroicon_outline "x-mark" aria_hidden="true" size=24 stroke_width=3 class="inline" %}
+    </button>
+
+    <!-- Menu -->
+    {% primary_navigation_items request.user as nav_items %}
+    <ul class="flex flex-col md:gap-8 p-4 md:p-0 md:flex-row" role="menubar">
+        {% for item in nav_items %}
+            <li
```
Comment (frjo) [inline hypha/core/templates/core/navigation/primarynav-apply.html:26]: Can we add a bit of padding to this `li` to make the focus ring visible?

File: hypha/static_src/sass/components/_nav.scss [hunk 3]
```diff
@@ -1,64 +1,18 @@
...
-
-        @include media-query(tablet-landscape) {
-            padding: 30px 0;
-        }
-
-        &--active {
+        @include media-query(md) {
             &::after {
                 position: absolute;
                 inset-block-end: 0;
```
Comment (frjo) [inline hypha/static_src/sass/components/_nav.scss:18]: I think I lean towards removing the `&--active` styling completely with this new design. With that I think we could completely remove this scss file and only use tailwind.

**Review by frjo — COMMENTED** [review]
I think this overall looks really nice!

Some very minor issues I commented on.

File: hypha/core/templates/core/navigation/primarynav-apply.html (hunk 1 above)
Comment (theskumar) [reply hypha/core/templates/core/navigation/primarynav-apply.html]: Done

File: hypha/core/templates/core/navigation/primarynav-apply.html (hunk 2 above)
Comment (theskumar) [reply hypha/core/templates/core/navigation/primarynav-apply.html:26]: It should work now.

File: hypha/static_src/sass/components/_nav.scss (hunk 3 above)
Comment (theskumar) [reply hypha/static_src/sass/components/_nav.scss:18]: I've not removed it yet. But I'll get to it before/after merging this PR, this shouldn't block testing right now.

File: hypha/templates/includes/user_menu.html [hunk 4]
```diff
@@ -1,67 +1,75 @@
 {% load i18n heroicons %}
 
 {% if request.user.is_authenticated %}
-    <div x-data="{ show: false }" class="relative flex">
+    <div x-data="{ show: false }" x-on:keydown.escape="show = false" class="relative flex">
         <button
             x-on:click="show = ! show"
-            class="font-bold flex items-center ps-1 pe-4 hover:bg-black hover:text-white transition-colors rounded-sm"
+            class="font-bold flex items-center transition-all rounded-sm px-1.5 bg-gray-100"
```
Comment (wes-otf) [inline hypha/templates/includes/user_menu.html:7]:
```suggestion
            class="font-bold flex items-center transition-all rounded-sm px-1.5 group"
```

For me the button slightly stands out against the background in a way that doesn't feel intentional, almost like it's a hover state
Screenshot 2024-07-03 at 12 19 04

Added `group` for the comment below on the user icon

File: hypha/templates/includes/user_menu.html [hunk 5]
```diff
@@ -1,67 +1,75 @@
...
             type="button"
+            :class="show ? 'bg-gray-900 text-white' : 'hover:bg-light-blue/30'"
```
Comment (wes-otf) [inline hypha/templates/includes/user_menu.html:9]:
```suggestion
            :class="show ? 'bg-gray-900 text-white' : 'hover:bg-slate-200'"
```

To match the notification bell hover state

File: hypha/templates/includes/user_menu.html [hunk 6]
```diff
@@ -1,67 +1,75 @@
...
         >
-            {% heroicon_micro "user-circle" class="inline align-text-bottom w-8 h-8 me-1" aria_hidden=true %}
-            <span>{{ request.user }}</span>
+            {% heroicon_micro "user-circle" class="inline align-text-bottom w-8 h-8 md:me-1" aria_hidden=true %}
```
Comment (wes-otf) [inline hypha/templates/includes/user_menu.html:11]:
```suggestion
            {% heroicon_micro "user-circle" class="inline group-hover:scale-110 transition-transform align-text-bottom w-8 h-8 md:me-1" aria_hidden=true %}
```
Just for more consistency with the notification bell, gives the cool icon scaling animation on hover.


https://github.com/HyphaApp/hypha/assets/145372368/dfb3966a-551b-4471-bd3e-93f405fabd54

**Review by wes-otf — COMMENTED** [review]
This is fantastic! Very slick and works really well. I tossed in a few super trivial changes but I think staff are really going to love this!

**Comment by wes-otf** [comment]
@frjo @theskumar the gray area mentioned above is still present, should I open another issue for this?

**Comment by theskumar** [comment]
> @frjo @theskumar the gray area mentioned above is still present, should I open another issue for this?

I'll get it done in another PR, another issue would be great.

**Pull request merged**